flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -125,4 +162,118 @@ public void validateColumnListParams(
+            RexNode simplifiedRexNode =
+                    FlinkRexUtil.simplify(
+                            sqlToRelConverter.getRexBuilder(),
+                            rexNode,
+                            relOptCluster.getPlanner().getExecutor());
+            List<RexNode> reducedNodes = new ArrayList<>();
+            relOptCluster
+                    .getPlanner()
+                    .getExecutor()
+                    .reduce(
+                            relOptCluster.getRexBuilder(),
+                            Collections.singletonList(simplifiedRexNode),
+                            reducedNodes);
+            // check whether period is the unsupported expression
+            if (!(reducedNodes.get(0) instanceof RexLiteral)) {
+                throw new UnsupportedOperationException(
+                        String.format(
+                                "Unsupported time travel expression: %s for the expression can not be reduced to a constant by Flink.",
+                                periodNode));
+            }
+
+            RexLiteral rexLiteral = (RexLiteral) (reducedNodes).get(0);
+            TimestampString timestampString = rexLiteral.getValueAs(TimestampString.class);
+            checkNotNull(
+                    timestampString,
+                    "The time travel expression %s can not reduce to a valid timestamp string. This is a bug. Please file an issue.",
+                    periodNode);
+
+            TableConfig tableConfig = ShortcutUtils.unwrapContext(relOptCluster).getTableConfig();
+            ZoneId zoneId = tableConfig.getLocalTimeZone();
+            long timeTravelTimestamp =
+                    TimestampData.fromEpochMillis(timestampString.getMillisSinceEpoch())
+                            .toLocalDateTime()
+                            .atZone(zoneId)
+                            .toInstant()
+                            .toEpochMilli();
+
+            SchemaVersion schemaVersion = TimestampSchemaVersion.of(timeTravelTimestamp);
+            IdentifierNamespace identifierNamespace = (IdentifierNamespace) ns;
+            IdentifierNamespace snapshotNameSpace =
+                    new IdentifierSnapshotNamespace(
+                            identifierNamespace,
+                            schemaVersion,
+                            ((DelegatingScope) usingScope).getParent());
+            ns = snapshotNameSpace;
+
+            sqlSnapshot.setOperand(
+                    1,
+                    SqlLiteral.createTimestamp(
+                            timestampString,
+                            rexLiteral.getType().getPrecision(),
+                            sqlSnapshot.getPeriod().getParserPosition()));
+        }
+
+        super.registerNamespace(usingScope, alias, ns, forceNullable);
+    }
+
+    /**
+     * For the time travel scenario, we need to build a SchemaVersion based on the Snapshot node for
```

Comment:
I think we don't need to talk time travel  in the java foc for this method.
I think may just be some thing like `Get the SqlSnapShot node in a SqlValidatorNamespace`.